pydis_site/apps/main/views/home.py

```diff
@@ -0,0 +1,101 @@
+
+        repo_dict = {repo_name: {} for repo_name in settings.HOMEPAGE_REPOS}
+
+        # Fetch the data from the GitHub API
+        api_data = requests.get(GITHUB_API)
+        api_data = api_data.json()
+
+        # Process the API data into our dict
+        for repo in api_data:
+            full_name = repo["full_name"]
+
+            if full_name in settings.HOMEPAGE_REPOS:
+                repo_dict[full_name] = {
+                    "full_name": repo["full_name"],
+                    "description": repo["description"],
+                    "language": repo["language"],
+                    "forks_count": repo["forks_count"],
+                    "stargazers_count": repo["stargazers_count"],
+                }
+        return repo_dict
+
+    def _get_repo_data(self):
+        """Build a list of RepositoryMetadata objects that we can use to populate the front page."""
+
+        # Try to get site data from the cache
+        try:
+            repo_data = RepositoryMetadata.objects.get(repo_name="python-discord/site")
+
+            # If the data is older than 2 minutes, we should refresh it.
+            if (timezone.now() - repo_data.last_updated).seconds > 120:
+
+                # Get new data from API
+                api_data_container = self._get_api_data()
+                repo_data_container = []
+
+                # Update or create all RepoData objects in settings.HOMEPAGE_REPOS
+                for repo_name, api_data in api_data_container.items():
+                    try:
+                        repo_data = RepositoryMetadata.objects.get(repo_name=repo_name)
+                        repo_data.description = api_data["description"]
+                        repo_data.language = api_data["language"]
+                        repo_data.forks = api_data["forks_count"]
+                        repo_data.stargazers = api_data["stargazers_count"]
+                    except RepositoryMetadata.DoesNotExist:
+                        repo_data = RepositoryMetadata(
+                            repo_name=api_data["full_name"],
+                            description=api_data["description"],
+                            forks=api_data["forks_count"],
+                            stargazers=api_data["stargazers_count"],
+                            language=api_data["language"],
+                        )
+                    repo_data.save()
+                    repo_data_container.append(repo_data)
+                return repo_data_container
+
+            # Otherwise, if the data is fresher than 2 minutes old, we should just return it.
+            else:
+                return RepositoryMetadata.objects.all()
+
+        # If this is raised, the database has no repodata at all, we will create them all.
+        except RepositoryMetadata.DoesNotExist:
+
+            # Get new data from API
+            api_data_container = self._get_api_data()
+            repo_data_container = []
```

Comment:
not a fan of these varnames, how about `api_repositories` and `database_repositories`